src/libraries/System.Data.Common/src/System/Xml/DataSetMappper.cs

```diff
@@ -314,13 +314,13 @@ internal bool IsRegionRadical(XmlBoundElement rowElem)
                 }
 
                 // must have no attributes
-                if (e.HasAttributes)
+                if (e!.HasAttributes)
```

Comment:
Why wasn't this failing before?

Reply:
It looks like the compiler used to infer `e != null` based on the call to `GetColumnSchemaForNode(rowElem, e!)`. Now, the compiler will only infer `e != null` when the warning that should be reported passing `e` to `GetColumnSchemaForNode(rowElem, e)` is not suppressed. See [sharplab.io](https://sharplab.io/#v2:C4LglgNgNAJiDUAfAxAOwK4QgQwEYQFMACA1PQgWACgABAJiJoEYB2agb2qO8aYDZGAFiIBZJgAoA9rgBWBAMbAA/EUkBKLj05Ueu3gAYVAZyIBeVUWwnm+gNya9ouuKMBCNfZ2OiAfTNEjADoAGVIAc2AAC09dAF8HXgEaYRFnaTlFVQ0vbm046ligA).